### PR TITLE
Add a new home directory for ChromeOS v90

### LINKED ISSUE #5400: Milestone 90 implements nosymfollow on home mount
More info: https://www.phoronix.com/scan.php?page=news_item&px=Linux-5.10-nosymfollow

This causes ruby problems if you have any gems installed because `xdg_base` symlinks 
`~/.config` to  `/usr/local/.config` and gem wants to drop local installs into that folder, which end up on a symlinked folder. 

This also causes issues with any apps which store preferences inside `~/.config` - which means all gtk apps.

```
chronos@localhost /usr/local/lib/crew/packages $ crew update
<internal:dir>:134:in `glob': Too many levels of symbolic links - /home/chronos/user/.local/share/gem/ruby/3.0.0/specifications (Errno::ELOOP)
        from /usr/local/lib64/ruby/site_ruby/3.0.0/rubygems/util.rb:101:in `glob_files_in_dir'
        from /usr/local/lib64/ruby/site_ruby/3.0.0/rubygems/specification.rb:771:in `gemspec_stubs_in'
        from /usr/local/lib64/ruby/site_ruby/3.0.0/rubygems/specification.rb:786:in `block in map_stubs'
        from /usr/local/lib64/ruby/site_ruby/3.0.0/rubygems/specification.rb:783:in `each'
        from /usr/local/lib64/ruby/site_ruby/3.0.0/rubygems/specification.rb:783:in `flat_map'
        from /usr/local/lib64/ruby/site_ruby/3.0.0/rubygems/specification.rb:783:in `map_stubs'
        from /usr/local/lib64/ruby/site_ruby/3.0.0/rubygems/specification.rb:776:in `installed_stubs'
        from /usr/local/lib64/ruby/site_ruby/3.0.0/rubygems/specification.rb:841:in `stubs_for_pattern'
        from /usr/local/lib64/ruby/site_ruby/3.0.0/rubygems/specification.rb:830:in `stubs_for'
        from /usr/local/lib64/ruby/site_ruby/3.0.0/rubygems/dependency.rb:276:in `matching_specs'
        from /usr/local/lib64/ruby/site_ruby/3.0.0/rubygems/dependency.rb:303:in `to_specs'
        from /usr/local/lib64/ruby/site_ruby/3.0.0/rubygems/dependency.rb:323:in `to_spec'
        from /usr/local/lib64/ruby/site_ruby/3.0.0/rubygems/core_ext/kernel_gem.rb:62:in `gem'
        from <internal:/usr/local/lib64/ruby/site_ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:73:in `require'
        from <internal:gem_prelude>:8:in `<internal:gem_prelude>'
```
One workaround is to run `sudo mount -o remount,symfollow /home/chronos/user` but a cleaner solution involves: 
- change `XDG_` variables not to store files in a symlinked folder.
- have ruby not store gems in a user folder by putting `gem: --no-user-install --no-rdoc --no-ri ` in `/usr/local/etc/gemrc`
- change the flatpak wrapper script to NOT use `--user`

### PR DESCRIPTION
**Pull request by supechicken**
Why not make a new home directory in `CREW_PREFIX` :)

Fixes #5400

Tested on `x86_64`

### CODE REVIEW

File: packages/unix_home.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  compatibility 'all'
+  license 'GPL-3+'
+  source_url 'file:///dev/null'
+  source_sha256 'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855'
+  
+  def self.install
+    @HOME = "#{CREW_DEST_PREFIX}/home/#{USER}/"
+    FileUtils.mkdir_p [@HOME, "#{CREW_DEST_PREFIX}/etc/env.d/"]
+    
+    FileUtils.ln_s Dir["#{Dir.home}/.bash*"], @HOME
```
Comment (uberhacker) [inline packages/unix_home.rb:15]: Isn't `#{Dir.home}` the same as `#{HOME}`?

File: packages/unix_home.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
+require 'package'
+
+class Unix_home < Package
```
Comment (uberhacker) [inline packages/unix_home.rb:3]: Is the idea here to make sure the home directory has execute permissions?

File: packages/unix_home.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    
+    FileUtils.ln_s Dir["#{Dir.home}/.bash*"], @HOME
+    FileUtils.ln_s "#{Dir.home}/Downloads", @HOME
+    FileUtils.ln_s "#{Dir.home}/MyFiles", @HOME + 'Documents'
+    FileUtils.ln_s Dir.home, @HOME + 'chromeos'
+    
+    @env = <<~EOF
+      set -a
+      HOME=#{@HOME}
+      ORIG_HOME=#{Dir.home}
```
Comment (uberhacker) [inline packages/unix_home.rb:23]: Maybe we should set crew constants also?

File: packages/unix_home.rb (hunk 1 above)
Comment (supechicken) [reply packages/unix_home.rb:15]: The `HOME` const will set to `CREW_PREFIX + Dir.home` if custom prefix set

File: packages/unix_home.rb (hunk 2 above)
Comment (supechicken) [reply packages/unix_home.rb:3]: Yes, and get rid of the `nosymfollow` mount symbol

File: packages/unix_home.rb (hunk 3 above)
Comment (supechicken) [reply packages/unix_home.rb]: Done

**Review by satmandu — APPROVED** [review]
Maybe we can queue this right after https://github.com/skycocker/chromebrew/pull/5692  so we don't have conflicts with `const.rb`, since this will need to be merged with master first...

**Comment by uberhacker** [comment]
@supechicken: This has conflicts.  Also, I would like to keep upfrog.sh just in case we might go back to Bintray.

File: packages/unix_home.rb [hunk 4]
```diff
@@ -0,0 +1,50 @@
+require 'package'
+
+class Unix_home < Package
+  description 'Regular Unix home directory for Chrome OS'
+  version '1.0'
+  compatibility 'all'
+  license 'GPL-3+'
+  source_url 'file:///dev/null'
+  source_sha256 'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855'
```
Comment (uberhacker) [inline packages/unix_home.rb:9]: Can we change to `source_url 'SKIP'` now that we have this new feature?

File: packages/unix_home.rb [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    
+    @env = <<~EOF
+      set -a
+      HOME=#{@HOME}
+      ORIG_HOME=#{HOME}
+      set +a
+    EOF
+    File.write("#{CREW_DEST_PREFIX}/etc/env.d/unix_home", @env)
+    # add original home directory to const.rb
+    system "echo 'ORIG_HOME=#{HOME}' >> #{CREW_LIB_PATH}/lib/const.rb"
```
Comment (uberhacker) [inline packages/unix_home.rb:28]: I don't think appending is a good idea.  Let's just set `ORIG_HOME` if the directory `#{CREW_DEST_PREFIX}/home/#{USER}/` exists and commit `const.rb`.

**Comment by supechicken** [comment]
I will reopen a new pr since I deleted the branch recently

**Pull request closed without merging**